Library/Formula/git.rb

```diff
@@ -11,34 +11,32 @@ class Git < Formula
   end
 
   resource "html" do
-    url "https://www.kernel.org/pub/software/scm/git/git-htmldocs-2.10.2.tar.xz"
-    sha256 "dd43111c3518a92a7fe64ac5cd32a5d4d77d49c67f7d89dce3e8293bc3d6b491"
+    url "https://www.kernel.org/pub/software/scm/git/git-htmldocs-2.40.1.tar.xz"
+    sha256 "8c08b31087566e719f6a7d16bb102255a8b9b970aefba6e306d6340eefe368ee"
   end
 
   resource "man" do
-    url "https://www.kernel.org/pub/software/scm/git/git-manpages-2.10.2.tar.xz"
-    sha256 "83b0a317f6039ad95ef6af6a182cf659c2d053eed5d8f70b06710eb787f8aa6f"
+    url "https://www.kernel.org/pub/software/scm/git/git-manpages-2.40.1.tar.xz"
+    sha256 "fe059c948ba3d169537b5b6b24f19726881057dfd4e5987f37789884d42fde13"
   end
 
   option "with-blk-sha1", "Compile with the block-optimized SHA1 implementation"
   option "without-completions", "Disable bash/zsh completions from 'contrib' directory"
-  option "with-brewed-curl", "Use Homebrew's version of cURL library" if MacOS.version > :snow_leopard
   option "with-brewed-svn", "Use Homebrew's version of SVN"
   option "with-persistent-https", "Build git-remote-persistent-https from 'contrib' directory"
 
   if MacOS.version == :tiger
     # system tar has odd permissions errors
     depends_on "gnu-tar" => :build
-    # Tiger's ld produces bad install-names for a keg-only curl
-    depends_on "ld64" => :build
-    depends_on "cctools" => :build
   end
 
   depends_on :expat
-  depends_on "pcre" => :optional
-  depends_on "gettext" => :optional
+  depends_on "pcre2" => :optional
+  depends_on "gettext"
   depends_on "openssl"
-  depends_on "curl" if MacOS.version < :snow_leopard || build.with?("brewed-curl")
+  depends_on "openssh" => :run
+  depends_on "curl"
+  depends_on "make" => :build
```

Comment:
Strangely, I'm seeing gmake fail at the `malloc` stage which is tanking the build for me. I wonder if I need to try building make with a different compiler.

Reply:
```
gmake(27946) malloc: ***  Deallocation of a pointer not malloced: 0x3029bc; This could be a double free(), or
free() called with the middle of an allocated block; Try setting environment variable MallocHelp to see tools
to help debug
gmake(27946) malloc: ***  Deallocation of a pointer not malloced: 0xa00064a0; This could be a double free(), o
r free() called with the middle of an allocated block; Try setting environment variable MallocHelp to see tool
s to help debug
gmake(27946) malloc: *** error for object 0x3029bc: pointer being reallocated was not allocated
gmake(27946) malloc: *** set a breakpoint in szone_error to debug
gmake: *** virtual memory exhausted
```

Reply:
hmm, I built gmake 4.4 using the stock GCC 4.0.1 on Tiger and used that.
Is there anything I need to do beyond `brew install -v git` to reproduce your scenario?

Reply:
I rebuilt with GCC 4.0 and it works now - I wonder if it was broken with GCC 4.2.

Reply:
behaviour is definitely weird. Do you get the same result if you build with GCC 4.2 on Tiger but turn off compiler optimisation in the formula?